src/App.tsx

```diff
@@ -15,49 +16,121 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  SORT_ALPHABETICALLY = 'Sort alphabetically',
+  SORT_BY_LENGTH = 'Sort by length',
+  REVERSE = 'Reverse',
+}
+
+interface Props {
+  sortField: string,
+  isReversed: boolean
+}
+
+const getPreparedGoods = (
+  goods: string[],
+  { sortField, isReversed }: Props,
+): string[] => {
+  const preapredGoods = [...goods];
+
+  if (sortField) {
+    preapredGoods.sort((good1: string, good2: string): number => {
+      switch (sortField) {
+        case SortType.SORT_ALPHABETICALLY: {
+          return good1.localeCompare(good2);
+        }
+
+        case SortType.SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default: {
+          return 0;
```

Comment:
Would be great to move switch...case out of the sort callback so there is no extra call if it's a default case